scripts/sct_dmri_display_bvecs.py

```diff
@@ -29,13 +31,27 @@
 def get_parser():
 
     # Initialize the parser
-    parser = Parser(__file__)
-    parser.usage.set_description('Display scatter plot of gradient directions from bvecs file.')
-    parser.add_option(name='-bvec',
-                      type_value='file',
-                      description='bvecs file.',
-                      mandatory=True,
-                      example='bvecs.txt')
+    parser = argparse.ArgumentParser(
+        description='Display scatter plot of gradient directions from bvecs file.',
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
```

Comment:
Do we want to keep the mandatory and optional argument discipline everywhere, or allow `argparse` to indicate what is required and what isn't?

Reply:
keeping these fields: I see pros (clarity from user standpoint) and cons (redundancy of logic, heavier code, in some functions what is optional/mandatory depends on some flags).

that's a good discussion topic for tmr's SCT meeting

Reply:
If we remove the argument groups but keep our convention of having everything be keyword arguments (e.g. `-bvecs`), argparse will try to group everything under optional arguments:

```bash
usage: sct_dmri_display_bvecs [-h] [-bvec <file>]

Display scatter plot of gradient directions from bvecs file.

optional arguments:
  -h, --help    show this help message and exit
  -bvec <file>  Input bvecs file. Example: bvecs.txt
```

This is even with `required=True`! I'm not sure there's a native way in argparse to have a "mandatory" keyword argument and have it be displayed properly. 

I think this is because the natural argparse-y way is `mandatory -> positional` and `optional -> keyword`. But, I'm not sure we want positional arguments... _(explicitly named keyword arguments seem important from a user perspective, considering the diversity of arguments we use across our many scripts.)_

Reply:
Also, I think because of this limitation, that's why we're adding `--help` manually, too.

Reply:
I could add comments explaining all of this. It definitely isn't standard usage, so it reads a bit funny when doing reviews like this. But, I think it might be an unavoidable consequence of using kwargs for both mandatory and optional arguments.